sdk/packages/core/src/api/validateOrder.test.ts

```diff
@@ -0,0 +1,107 @@
+import { testAccount, testOrder } from '@omni-network/test-utils'
+import { expect, test, vi } from 'vitest'
+import * as api from '../internal/api.js'
+import { validateOrder } from './validateOrder.js'
+
+// TODO calls as empty array should not be allowed // throw error
+
+test('default: native transfer order', async () => {
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue({
+    accepted: true,
+  })
+
+  const resultPromise = validateOrder('http://localhost', {
+    owner: testAccount.address,
+    srcChainId: 1,
+    destChainId: 2,
+    calls: [
+      {
+        target: testAccount.address,
+        value: 0n,
+      },
+    ],
+    deposit: {
+      amount: 0n,
+    },
+    expense: {
+      amount: 0n,
+    },
+  })
+  await expect(resultPromise).resolves.toEqual({ accepted: true })
+  expect(api.fetchJSON).toHaveBeenCalledWith(
+    'http://localhost/check',
+    expect.any(Object),
+  )
+})
+
+test('default: order', async () => {
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue({
+    accepted: true,
+  })
+  await expect(validateOrder('http://localhost', testOrder)).resolves.toEqual({
+    accepted: true,
+  })
+  expect(api.fetchJSON).toHaveBeenCalledWith(
+    'http://localhost/check',
+    expect.any(Object),
+  )
+})
+
+test('behaviour: resolves if response is supported error object', async () => {
+  const response = {
+    error: {
+      code: 1,
+      message: 'an error',
+    },
+  }
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue(response)
+  await expect(validateOrder('http://localhost', testOrder)).resolves.toBe(
+    response,
+  )
```

Comment:
do we want to resolve on supported error? or throw on supported error.

my understanding is that react query pattern is to throw on errors, so you can use their error states

Reply:
Ideally yes, but again I wanted to keep the core aligned to the react package's expectations, which handles the `error` object from the response. I think we need a bigger refactoring of error handling, especially having clear differences between input errors, known server errors and possible exceptions.